### PR TITLE
Efficient falseToFailure

### PR DESCRIPTION
**Pull request by TysonMN**

### CODE REVIEW

**Comment by AlexeyRaga** [comment]
This is great as it now works as expected!

**Review by dharmaturtle — COMMENTED** [review]
Just wondering, why does `PropertyBuilder`'s `Return` still use `Property.ofBool`?

**Comment by TysonMN** [comment]
> why does `PropertyBuilder`'s `Return` still use `Property.ofBool`?

I think that is dead code.

File: src/Hedgehog/Property.fs [hunk 1]
```diff
@@ -95,7 +95,7 @@ module Property =
         |> ofGen
 
     let falseToFailure p =
-        p |> bind ofBool
+        p |> map (fun b -> if not b then failwith "Expected 'true' but was 'false'.")
```
[the remark this answers is not in the document]
Comment (TysonMN) [reply src/Hedgehog/Property.fs]: > Should there also be a custom exception for this?

Yes, good points.  I added your code to Hedgehog and raised that exception instead.

> Should we rename the function? Nothing's coming immediately to mind for me...

We previous discussed this, and the name `falseToFailure` was the best we came up with at the time.  We can continue to think of a better name, but this isn't part of this PR.

> P.S. What does `> Property.recheck "0_13250624720205828615_16980185048476617913_" <property>` mean? In particular, the first `>` and the last `<property>`. Always made me 🤨

It means that code like
```fsharp
Property.check myProperty
```

should be changed to
```fsharp
Poperty.recehck "0_13250624720205828615_16980185048476617913_" myProperty
```

to recheck the value the just failed.  I am open to improving this message.  Also not part of this PR.

**Pull request merged**